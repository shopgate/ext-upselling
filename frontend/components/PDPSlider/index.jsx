import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Slider from '../Slider';
import { makeConnectProductWithRelations } from '../connectors';

/**
 * PDPSlider component.
 * @returns {JSX}
 */
const PDPSlider = ({ productId, config }) => {
  if (!productId) {
    return null;
  }

  if (!config) {
    return null;
  }

  return (
    <Slider
      headline={config.headline}
      type={config.type}
      productId={productId}
      showPrice={config.showPrice || false}
      showName={config.showName || false}
      titleRows={config.nameLines || 2}
      property={config.property}
    />
  );
};

PDPSlider.propTypes = {
  config: PropTypes.shape({
    headline: PropTypes.string,
    nameLines: PropTypes.number,
    property: PropTypes.string,
    showName: PropTypes.bool,
    showPrice: PropTypes.bool,
    titleRows: PropTypes.number,
    type: PropTypes.string.isRequired,
  }),
  productId: PropTypes.string,
};

PDPSlider.defaultProps = {
  productId: null,
  config: {},
};

export default makeConnectProductWithRelations()(memo(PDPSlider));
