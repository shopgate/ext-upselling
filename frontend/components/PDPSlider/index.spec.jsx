import React from 'react';
import { mount } from 'enzyme';
import getConfig from '../../helpers/getConfig';
import PDPSlider from './index';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  memo: cmp => cmp,
}));
let mockedShowPrice = true;
let mockedShowName = true;
jest.mock('../../helpers/getConfig', () => () => ({
  productPage: [
    {
      type: 'mockedType',
      headline: 'mockedHeadline',
      get showPrice() { return mockedShowPrice; },
      get showName() { return mockedShowName; },
    },
  ],
}));
jest.mock('../Slider', () => function Slider() {
  return null;
});
jest.mock('../connectors', () => ({
  makeConnectProductWithRelations: () => PDPSheetComponent => PDPSheetComponent,
}));

describe('PDPSlider', () => {
  it('should render with price and names', () => {
    const component = mount(<PDPSlider productId="mockedProductId" config={getConfig().productPage[0]} />);

    expect(component.find('PDPSlider').props().productId).toBe('mockedProductId');
    expect(component.find('Slider').props()).toMatchObject(getConfig().productPage[0]);
    expect(component).toMatchSnapshot();
  });

  it('should render without price and names as default', () => {
    mockedShowPrice = null;
    mockedShowName = null;
    const component = mount(<PDPSlider productId="mockedProductId" config={getConfig().productPage[0]} />);

    expect(component.find('Slider').props().showPrice).toBe(false);
    expect(component.find('Slider').props().showName).toBe(false);
    expect(component).toMatchSnapshot();
  });

  it('should render nothing when productId is not ready', () => {
    const component = mount(<PDPSlider config={getConfig().productPage[0]} />);
    expect(component.html()).toBe(null);
  });
});
