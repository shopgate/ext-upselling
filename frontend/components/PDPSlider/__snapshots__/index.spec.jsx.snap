// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDPSlider should render with price and names 1`] = `
<PDPSlider
  config={
    Object {
      "headline": "mockedHeadline",
      "showName": true,
      "showPrice": true,
      "type": "mockedType",
    }
  }
  productId="mockedProductId"
>
  <Slider
    headline="mockedHeadline"
    productId="mockedProductId"
    showName={true}
    showPrice={true}
    titleRows={2}
    type="mockedType"
  />
</PDPSlider>
`;

exports[`PDPSlider should render without price and names as default 1`] = `
<PDPSlider
  config={
    Object {
      "headline": "mockedHeadline",
      "showName": null,
      "showPrice": null,
      "type": "mockedType",
    }
  }
  productId="mockedProductId"
>
  <Slider
    headline="mockedHeadline"
    productId="mockedProductId"
    showName={false}
    showPrice={false}
    titleRows={2}
    type="mockedType"
  />
</PDPSlider>
`;
